### PR TITLE
feat(mongo_translator): Implemented convert step [TCTC-1832]

### PR DESCRIPTION
**Pull request by fspot**
Translates the concatenate step from `../src/lib/translators/mongo.ts`

:warning: I had to handle some edge cases in order to make it iso with the pandas executor. There are probably still missing cases because type conversion in mongo and pandas have different rules. I also fear that adding these edge cases might decrease performances.

### CODE REVIEW

**Review by Sanix-Darker — APPROVED** [review]
LGTM !

File: server/tests/backends/mongo_translator/test_mongo_translator_steps.py [hunk 1]
```diff
@@ -69,5 +70,11 @@ def test_mongo_translator_pipeline(mongo_database, case_id, case_spec_file_path)
     df = pd.DataFrame(result)
     if '_id' in df:
         df.drop('_id', axis=1, inplace=True)
-    expected_df = pd.read_json(json.dumps(spec['expected']), orient='table')
+    expected_df = pd.read_json(
+        json.dumps(
+            spec[f'expected_{exec_type}' if f'expected_{exec_type}' in spec else 'expected']
```
Comment (davinov) [inline server/tests/backends/mongo_translator/test_mongo_translator_steps.py:75]: I guess this is to have specific expectations for mongo in fixtures? But I don't see `expected_mongo` property anywhere. Are they created by the `retrieve_case` function?
If yes, have I have miss any mongo specific declaration in the JSON fixtures?

File: server/src/weaverbird/backends/mongo_translator/steps/convert.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
+from typing import List, Optional, Union
+
+from weaverbird.backends.mongo_translator.steps.types import MongoStep
+from weaverbird.pipeline.steps import ConvertStep
+
+TYPE_MAP = {
+    'boolean': 'bool',
+    'date': 'date',
+    'float': 'double',
+    'integer': 'long',  # better to cast into 64-bit integers. 32-bit integers ('int') cannot be casted to dates
```
Comment (davinov) [inline server/src/weaverbird/backends/mongo_translator/steps/convert.py:10]: :+1:

**Review by davinov — COMMENTED** [review]
This is an interesting step to port!
Today, we have not the same results for the convert step in mongo and in pandas. Question is: do we want to have exact matching results, or is it OK if we have specific fixtures for this step for each backend.
I have no strong opinion on this :shrug:

File: server/tests/backends/mongo_translator/test_mongo_translator_steps.py (hunk 1 above)
Comment (fspot) [reply server/tests/backends/mongo_translator/test_mongo_translator_steps.py:75]: I modified this code because I thought I would end up adding a `expected_mongo` property, but in the end I managed to get the same results than with pandas.

File: server/src/weaverbird/backends/mongo_translator/steps/convert.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                        # Mongo cannot cast integers into date but only long into date,
+                        # so we manage the cast from int to long when needed:
+                        {
+                            'case': {'$eq': [{'$type': f'${column}'}, 'int']},
+                            'then': {'$toLong': f'${column}'},
+                        },
+                    ],
+                    'default': f'${column}',
+                }
+            }
```
Comment (davinov) [inline server/src/weaverbird/backends/mongo_translator/steps/convert.py:46]: Since last version, we use `todate` step instead of the `convert` step when someone tries to cast a column to date types. This makes this branch less critical to implement exactly right.

File: server/src/weaverbird/backends/mongo_translator/steps/convert.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        elif mongo_type == 'string':
+            # First, convert NaN to null:
+            _input = {
+                '$switch': {
+                    'branches': [{'case': {'$eq': [f'${column}', float('NaN')]}, 'then': None}],
+                    'default': f'${column}',
+                }
+            }
+            # Then, null will be converted to the literal string 'nan':
+            _on_null = 'nan'
```
Comment (davinov) [inline server/src/weaverbird/backends/mongo_translator/steps/convert.py:74]: This is a specific pandas' behavior to use the string 'nan'.
Maybe we would think than `null` / `None` is more appropriate? (because we can filter on it quickly after, and it denotes some missing data)
Maybe that's something we'd like to do differently than pandas?

File: server/tests/backends/mongo_translator/test_mongo_translator_steps.py (hunk 1 above)
Comment (davinov) [reply server/tests/backends/mongo_translator/test_mongo_translator_steps.py:75]: :ok_hand: I was afraid of missing something, I understand better now how this ended up there :+1:

**Review by davinov — APPROVED** [review]
:+1:
